packages/ui/src/components/form-input/components/form-text-input.tsx

```diff
@@ -36,13 +22,10 @@ const FormTextInput = forwardRef<HTMLInputElement, FormTextInputPropsType>((prop
       name={props.name}
       control={formContext.control}
       render={({ field, fieldState }) => {
-        // Use proper React Hook Form reference handling
-        field.ref = setRefs
-
         // form error takes precedence over props.error
-        const errorMessage = fieldState.error?.message || props.error
+        const error = fieldState.error?.message ?? props.error
```

Comment:
Why change from `||` to `??`. The only real difference is to _allow_ the display of `0`, `false` and `''` to override an error deliberately passed in. Seems that we should only care about falsiness, not where it's _specifically_ `null` or `undefined`.